server/src/main/java/org/opensearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -1060,13 +1129,27 @@ private void doDeleteShardSnapshots(
                 ActionListener.wrap(() -> listener.onResponse(updatedRepoData)),
                 2
             );
+
+            // We can create map of indexId to ShardInfo based on the old repository data. This is later used in cleanup
+            // of stale indexes in combination with Snapshot Shard Paths file
```

Comment:
when would this additional info over snapshot shard paths file ?

Reply:
Snapshot shard paths file upload is an essential part during finalise snapshot. However, we don't fail snapshots if these upload fails. Hence we also add the information about the IndexIds and their shards from the old repository data. This was caught in one of the tests that failed during the integ test run with hashed_prefix approach.